### ISSUE TITLE
how to get `Object.keys(<type>)` with new api version?

### ISSUE THREAD

**Issue by carlosala** [polkadot-js/api#5131]
Hi @jacogr!
First of all, sorry for not following the issue template, it's a short question and I think it's going to be easier like that!

I would like to have the same approach as I had before v9 of polkadot-js of `Object.keys(<type>)` to get all the fields available on that. How can I get that with the `Map` approach?
Thanks!

**Comment by carlosala** [comment]
I don't know why, but I can't get `.keys()`, `.entries()` and `.toArray()` work. `.defKeys` works, thanks!
I probably have something not properly defined, but it looks like I'm not getting a Map.

**Comment by carlosala** [comment]
I'll look what's actually going on, but I can stay for the moment in `v8` until I get everything working 👌🏻

**Comment by jacogr** [comment]
Anyway, `.keys()` & `.entries()` in JS Map are iterators, so they are, imho, a bit more fiddly, i.e. -

```js
// this is an iterator, not an array
for (const [k, v] of obj.entries()) {
 ...
}

// convert iterator to array of keys and map
[...obj.keys()];
  .map((k) => {
    const v = obj[k];
    ...
  });
```

As for `toArray`, just realized by looking that it only returns the values, not a k/v pair... Sorry about that comment - Where `toArray` is not working, it [should do so on Struct](https://github.com/polkadot-js/api/blob/41925a9893e5d8e93b574fad8c60157a76f22a54/packages/types-codec/src/native/Struct.ts#L262-L264)).

**Comment by carlosala** [comment]
I finally solved the issue in a similar way. When working with enums, we were really dependent on the following structure:
```typescript
for (const isKey of Object.keys(type).filter(x => x.startsWith('is')) {
  if (type[isKey]) {
    blablabla
  }
}
```
Then I replaced that `Object.keys` with `type.defKeys.map((v: string) => 'is' + v)`. It's a bit tricky and not really understandable, but at least I left a decent comment in the code.
Thanks for all your suggestions, were really useful.
As always, you rock! 🚀 💪🏻
